cli/src/merge_tools/builtin.rs

```diff
@@ -64,6 +64,27 @@ pub struct FileInfo {
     contents: FileContents,
 }
 
+impl FileInfo {
+    // Returns `Some(true)` if the file exists and is empty.
+    // Returns `None` if the file is absent and does not exist.
+    fn is_empty(&self) -> Option<bool> {
+        if let FileContents::Absent = self.contents {
+            // If the contents are absent the mode should be absent as well.
+            debug_assert!(self.file_mode == scm_record::FileMode::absent())
+        }
+        match self.contents {
+            // Being absent is different from being present and empty.
+            FileContents::Absent => None,
```

Comment:
```suggestion
            FileContents::Absent => {
                // If the contents are absent the mode should be absent as well.
                debug_assert!(self.file_mode == scm_record::FileMode::absent());
                None
            }
```

suggestion: Inline the assertion here?

Reply:
Good point, thanks!